### ISSUE TITLE
flutter_attach integration test is flaky

### ISSUE THREAD

**Issue by DanTup** [flutter/flutter#20822]
Seen this a couple of times while running all the tests on my fork trying to repro other flakiness.

We try to send `stop` to both the `run` instance and the `attach` instance. However the second one often fails saying it can't find the app (which I think is valid, because they were the same running app, so sending a stop should stop!).

```
[+ 38597]    [{"event":"app.start","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","deviceId":"flutter-tester","directory":null,"supportsRestart":true}}]
[+ 39036]    Error 1001 received from application: File system already exists
[+ 39062]    [{"event":"app.debugPort","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","port":49446,"wsUri":"ws://127.0.0.1:49446/ws","baseUri":"file:///var/folders/xb/469gd9j941bc7v3mltn7zgb40000gn/T/flutter_attach_test.J5PDkPzZS2gg/flutter_attach_test.J5PDkP/"}}]
[+ 40921]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"0","progressId":null,"message":"Syncing files to device Flutter test device..."}}]
[+ 51887]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"0","progressId":null,"finished":true}}]
[+ 51913]    [{"event":"app.started","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627"}}]
[+ 51920]    [{"id":1,"method":"app.restart","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","fullRestart":false,"pause":false}}]
[+ 51932]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"1","progressId":"hot.reload","message":"Initializing hot reload..."}}]
[+ 52938]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"2","progressId":null,"message":"Syncing files to device Flutter test device..."}}]
[+ 53139]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"2","progressId":null,"finished":true}}]
[+ 53498]    [+18374 ms] topLevelFunction
[+ 53664]    [{"event":"app.progress","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627","id":"1","progressId":"hot.reload","finished":true}}]
[+ 53668]    Reloaded 0 of 403 libraries in 1,732ms.
[+ 53670]    [{"id":1,"result":{"code":0,"message":"Reloaded 0 of 403 libraries"}}]
[+ 53687]    Closing VM service
[+ 53698]    Stopping app
[+ 53698]    [{"id":1,"method":"app.stop","params":{"appId":"91f5f452-f3ec-4b68-9980-99547ea191f5"}}]
[+ 53713]    [ +215 ms] DevFS: Deleting filesystem on the device (file:///var/folders/xb/469gd9j941bc7v3mltn7zgb40000gn/T/flutter_attach_test.J5PDkPt4kxeI/flutter_attach_test.J5PDkP/)
[+ 53715]    [   +1 ms] _deleteDevFS: {fsName: flutter_attach_test.J5PDkP}
[+ 53722]    [   +6 ms] DevFS: Deleted filesystem on the device (file:///var/folders/xb/469gd9j941bc7v3mltn7zgb40000gn/T/flutter_attach_test.J5PDkPt4kxeI/flutter_attach_test.J5PDkP/)
[+ 53727]    [   +5 ms] ext.flutter.exit: {isolateId: isolates/232555921}
[+ 53783]    [  +42 ms] Service protocol connection closed.
[+ 53784]    Lost connection to device.
[+ 53966]    [ +195 ms] Application finished.
[+ 53967]    [{"id":1,"result":true}]
[+ 53968]    Waiting for process to end
[+ 53970]    [{"event":"app.stop","params":{"appId":"91f5f452-f3ec-4b68-9980-99547ea191f5"}}]
[+ 54003]    [  +36 ms] "flutter run" took 52,412ms.
[+ 54079]    Process exited (0)
[+ 54079]    Stopping app
[+ 54080]    [{"id":2,"method":"app.stop","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627"}}]
[+ 54092]    [{"event":"app.stop","params":{"appId":"a13fe6c4-f623-47ef-8243-e4ae024f0627"}}]
[+ 54116]    [{"id":2,"error":"app 'a13fe6c4-f623-47ef-8243-e4ae024f0627' not found","trace":"#0      AppDomain.stop (package:flutter_tools/src/commands/daemon.dart:508:7)\n<asynchronous suspension>\n#1      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#2      new Future.sync (dart:async/future.dart:224:31)\n#3      Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#4      Daemon._handleRequest (package:flutter_tools/src/commands...
53:00 +434 ~1 -4: test/integration/flutter_attach_test.dart: attached process can hot reload [E]                                                                                                       
  Unexpected error response
  [{"id":2,"error":"app 'a13fe6c4-f623-47ef-8243-e4ae024f0627' not found","trace":"#0      AppDomain.stop (package:flutter_tools/src/commands
```

Maybe easier to understand with the app ID highlighted:

![screen shot 2018-08-20 at 5 47 45 pm](https://user-images.githubusercontent.com/1078012/44354233-358e2c80-a4a1-11e8-8720-0d4e2fa38562.png)

**Comment by DanTup** [comment]
Seen again in https://cirrus-ci.com/task/4673346587328512

**Comment by DanTup** [comment]
This should've been fixed by the swap to detaching correctly (#21154). If anyone sees it again, lmk.
